### DISCUSSION TITLE
Medienbrowser: Dialog sollte alle Beiträge verlinken, wenn die ausgewählten Dateien, welche gelöscht werden möchten, noch in Verwendung sind.

### DISCUSSION THREAD

**Discussion by techotline** [lotta-schule/web#232] [Ideen]
### Optimierung des Medienbrowsers

Beiträge mit der Datei anzeigen, wenn diese im Medienbrowser gelöscht werden möchte.

### Problem
Es befindet sich eine Datei in meinem privaten Medienbrowser, die ich unkompliziert entfernen möchte, weil sie entweder nicht mehr aktuell ist oder benötigt wird. Es erscheint mir daher sinnvoll, _**alle**_ Referenzen einzusehen, in der genau diese Datei (z.B. ein Gruppenfoto der Klasse 7/3) in Verwendung ist. 

**Beispiel Klickweg:** Meine Dateien und Medien -> File auswählen -> "Rechtsklick" -> Löschen

### Darstellung

mobil view:

![deleteFileswithDialogandArticles](https://github.com/lotta-schule/web/assets/153733311/e069bb37-6748-42b6-9e25-722c3bf0b32a)

desktop:

<img width="1440" alt="web_deleteFileswithDialogandArticles" src="https://github.com/lotta-schule/web/assets/153733311/5cf486be-612b-458b-85e4-37fea7b00a8a">

### Idee

Dialog öffnet sich und nennt alle Beiträge, in der sich diese Datei befindet, weil diese Beiträge ansonsten ohne jene Datei veröffentlicht sind.

**Warum + Beispiel**

Nutzer löschen Dokumente und bemerken nicht, dass ihre zuvor erstellten Beiträge nur noch Text beinhalten, sofern der Beitrag keine Galerie ist. Insbesondere Vorschaubilder werden mehrfach gebraucht und vielseitig eingesetzt, deren Verwendung man nur bedingt nachvollziehen kann.

<img width="1221" alt="fb_dateilos" src="https://github.com/lotta-schule/web/assets/153733311/5ad8ab51-4c50-4e52-8c77-bd2609b2f498">

**Comment by wie0815** [comment]
Das Problem ist sehr komplex. Ich hätte die Idee, dass Bild, Video oder Ton-Module, die keine Datei anzeigen, da keine verlinkt bzw. die Datei gelöscht wurde, nicht angezeigt werden.
Ist das machbar @ptitmouton?

**Reply by ptitmouton** [reply]
Ok @wie0815 ich hab zwei Mal lesen gebraucht um zu verstehen, was du meinst, da ist viel impliziter Zusammenhang zu @techotline Vorschlag, ich denke die zwei Probleme haben nur vage miteinander zu tun.

@wie0815 möchte die Konsequenz aus dem Löschen von benutzten Dateien minimieren, indem Module, die hauptsächlich da sind, um eine bestimmte Datei anzuzeigen, nicht angezeigt werden wenn keine Datei eingestellt ist.
Das ist schnell erledigt und ich halte es für eine sehr gute, pragmatische Idee. Sollten wir so machen. 👍🏻

@techotline möchte die vermeintliche Ursache lösen: Viele dieser Dateien würden gelöscht, ohne dass der Nutzer gewusst hätte, dass sie eingebunden sind.
Grundsätzlich auch sehr sinnvoll, zumal wir die Möglichkeit backendseitig ja schon haben, abzufragen wo eine Datei genutzt wird.
Das ist eher ein Darstellungsproblem als alles andere:
Eine Datei löschen könnte eine Liste anzeigen, wo diese Datei verwendet wird. Soweit so klar.
Wie gehen wir damit um, wenn nicht-leere Ordner gelöscht werden sollen? Wir sollten uns ein Interface überlegen, dass dem Nutzer die Verwendung von *vielen* Dateien vor der Löschung präsentieren zur Bestätigung präsentieren kann.
Wenn wir das hätten, könnten wir auch einfach unseren Löschprozess etwas vereinheitlichen (ähnlicher oder gleicher für einzelne oder mehrere Dateien, einzelne oder mehrere Ordner oder sogar Dateien/Ordner gemischt.)
Ich denke die Schwierigkeit liegt darin, ein übersichtliches verständliches UI dafür zu bauen (heißt: Ich will zuerst Mockup-Vorschläge von @BillyTheKing )

**Reply by BillyTheKing** [reply]
Ich kümmere mich um Mockups

**Comment by BillyTheKing** [comment]
![Web – Neuer Medienbrwoser – Löschen Dialog@2x](https://github.com/lotta-schule/web/assets/11082197/304addc5-bb38-49de-b7d0-5677238375f2)

**Kurze Erklärung:**
Werden mehrere Dateien oder Dateien und Ordner zum Löschen ausgewählt, werden alle zu löschenden Dateien, also auch alle sich innerhalb eines oder mehrerer Ordner befindlicher Dateien, die an irgendeiner Stelle im Lotta verwendet werden, aufgelistet. Mit einem Klick auf den kleinen Pfeil wird eine Auflistung der entsprechenden Stelle(n) jeder Datei anzeigt. Wird eine neue Datei zur Überprüfung ausgewählt, wird die andere Auflistung der Stellen, gelb hinterlegt, eingeklappt (@ptitmouton). 

Eine weitere Aktion ist nicht vorgesehen. Der User bestätigt anschließend mit einem Klick auf den Button.

* in der mobilen Ansicht entfällt die gelb hinterlegte Einrückung.

**Comment by ptitmouton** [comment]
Denke Problemstellung ist klar geworden und von  #246  und #247 abgedeckt
